Clarify cluster diagram as one possible reference architecture (#47164)
kubernetes/website: content/en/docs/concepts/overview/components.md

[kube-apiserver](/docs/concepts/architecture/#kube-apiserver)
: The core component server that exposes the Kubernetes HTTP API

[etcd](/docs/concepts/architecture/#etcd)
: Consistent and highly-available key value store for all API server data

[kube-scheduler](/docs/concepts/architecture/#kube-scheduler)
: Looks for Pods not yet bound to a node, and assigns each Pod to a suitable node.

[kube-controller-manager](/docs/concepts/architecture/#kube-controller-manager)
: Runs {{< glossary_tooltip text="controllers" term_id="controller" >}} to implement Kubernetes API behavior.

[cloud-controller-manager](/docs/concepts/architecture/#cloud-controller-manager) (optional)
: Integrates with underlying cloud provider(s)

### Node Components

Run on every node, maintaining running pods and providing the Kubernetes runtime environment:

[kubelet](/docs/concepts/architecture/#kubelet)
: Ensures that Pods are running, including their containers.

[kube-proxy](/docs/concepts/architecture/#kube-proxy) (optional)
: Maintains network rules on nodes to implement {{< glossary_tooltip text="Services" term_id="service" >}}

[Container runtime](/docs/concepts/architecture/#container-runtime)

> : Software responsible for running containers. Read [Container Runtimes](/docs/setup/production-environment/container-runtimes/) to learn more. {{% thirdparty-content single="true" %}}

Reviewer: ```suggestion
: Software responsible for running containers. Read [Container Runtimes](/docs/setup/production-environment/container-runtimes/) to learn more.

{{% thirdparty-content single="true" %}}
```
It _precedes_ the content that it applies to.

Your cluster may require additional software on each node; for example, you might also

> run systemd on a Linux node to supervise local components.

Reviewer: On reflection
```suggestion
run [systemd](https://systemd.io/) on a Linux node to supervise local components.
```

---

<!-- overview -->

This page provides a high-level overview of the essential components that make up a Kubernetes cluster.

{{< figure src="/images/docs/components-of-kubernetes.svg" alt="Components of Kubernetes" caption="The components of a Kubernetes cluster" class="diagram-large" clicktozoom="true" >}}

<!-- body -->

## Core Components

A Kubernetes cluster consists of a control plane and one or more worker nodes. Here's a brief overview of the main components:

### Control Plane Components

Manage the overall state of the cluster:

> <dl>
>   <dt>kube-apiserver</dt>
>   <dd>The API server that exposes the Kubernetes API</dd>
> 
>   <dt>etcd</dt>
>   <dd>Consistent and highly-available key value store for all cluster data</dd>
> 
>   <dt>kube-scheduler</dt>
>   <dd>Assigns Pods to Nodes</dd>
> 
>   <dt>kube-controller-manager</dt>
>   <dd>Runs controller processes</dd>
> 
>   <dt>cloud-controller-manager</dt>
>   <dd>Integrates with underlying cloud providers</dd>
> </dl>

Reviewer: /docs/concepts/architecture/#kube-apiserver
```suggestion
[kube-apiserver](/docs/concepts/architecture/#kube-apiserver)
: The core component server that exposes the Kubernetes HTTP API

[etcd](/docs/concepts/architecture/#etcd)
: Consistent and highly-available key value store for all API server data

[kube-scheduler](/docs/concepts/architecture/#kube-scheduler)
: Looks for Pods not yet bound to a node, and assigns each Pod to a suitable node.

[kube-controller-manager](/docs/concepts/architecture/#kube-controller-manager)
: Runs {{< glossary_tooltip text="controllers" term_id="controller" >}} to implement Kubernetes API behavior.

[cloud-controller-manager](/docs/concepts/architecture/#cloud-controller-manager) (optional)
: Integrates with underlying cloud provider(s)
```

Author: thanks for the tip @sftim, I think there might be other sections that need converting too, I'll fix them in a separate commit!

### Node Components

Run on every node, maintaining running pods and providing the Kubernetes runtime environment:

> <dl>
>   <dt>kubelet</dt>
>   <dd>Ensures that containers are running in a Pod</dd>
> 
>   <dt>kube-proxy</dt>
>   <dd>Maintains network rules on nodes</dd>
> 
>   <dt>Container runtime</dt>
>   <dd>Software responsible for running containers</dd>
> </dl>

Reviewer: ```suggestion
kubelet
: Ensures that Pods are running, including their containers.

kube-proxy (optional)
: Maintains network rules on nodes to implement {{< glossary_tooltip text="Services" term_id="service" >}}

Container runtime
: Software responsible for running containers. Read [Container Runtimes](/docs/setup/production-environment/container-runtimes/) to learn more.
```

Author: It seems gh doesn't allow joint commits for suggestions with multi-line deletes, so I'll batch the rest of the changes in the follow up commit!

> 

Reviewer: _Optional_
```suggestion

{{% thirdparty-content single="true" %}}

Your cluster may require additional software on each node; for example, you might also
run systemd on a Linux node to supervise local components.

```

> ## Addons

Reviewer 2: ```suggestion
## Add-ons
```
I don't know why this suggestion box extends into the two editor layouts.

Reviewer: This suggestion is not part of the change that the PR description is summarizing, so @robert-cronin you are welcome to skip it.

Author: @network-charles I'll skip this change to maintain consistency with current docs as discussed above. Thanks for flagging it.

Reviewer 2: Ok

> Addons extend the functionality of Kubernetes. A few important examples include:

Reviewer 2: ```suggestion
Add-ons extend the functionality of Kubernetes. A few important examples include:
```

Reviewer: This suggestion is not part of the change that the PR description is summarizing, so @robert-cronin you are welcome to skip it.

Author: @network-charles I'll skip this change to maintain consistency with current docs as discussed above. Thanks for flagging it.
